File: game_state.cpp
#include "game_state.h"

namespace game_state
{
    bool game_running = true;

    // Struct constructors
    game_state::Region::Region(std::string rname, std::string rlevel_name, int rlevel_num) : name(rname), level_name(rlevel_name), level_num(rlevel_num) {}
    game_state::Player::Player(std::string pname, std::string ptag, int page, int ppronouns, int pregion, int padvantage, int pdisadvantage, int pneutral, int pmental) 
        : name(pname), tag(ptag), age(page), pronouns(ppronouns), region(pregion), advantage(padvantage), disadvantage(pdisadvantage), neutral(pneutral), mental(pmental) {}

    std::string player_name = "";
    std::string player_tag = "";
    int player_pronouns = -1;
    int player_region = -1;

    std::unique_ptr<Player> player = std::make_unique<Player>(
        player_name, player_tag, 13, player_pronouns, player_region, 0, 0, 0, 0
    );

    std::vector<Region> Regions = 
    {
        Region{"PADDING", "FOR CONSISTENT INDEXING", -1},
        Region{"Japan", "Very Competitive", 5},
        Region{"NA East", "Competitive", 4},
        Region{"China", "Competitive", 4},
        Region{"UK/Ireland", "Hard", 4},
        Region{"South Korea", "Hard", 4},
        Region{"EU West", "Moderate", 3},
        Region{"NA West", "Moderate", 3},
        Region{"NA Midwest", "Moderate", 3},
        Region{"South America", "Moderate", 3},
        Region{"Europe North/East", "Light", 2},
        Region{"Mexico", "Light", 2},
        Region{"Middle East", "Light", 2},
        Region{"Asia Southeast", "Obscure", 1},
        Region{"Asia South", "Obscure", 1},
        Region{"Central America", "Obscure", 1},
        Region{"Africa", "Obscure", 1},
        Region{"Oceaina", "Obscure", 1}
    };

    std::unordered_map<int, std::array<std::string, 2>> Pronouns = 
    {
        {1, {"he", "him"}},
        {2, {"she", "her"}},
        {3, {"they", "them"}}
    };
}

File: Python (old)/game_state.py
class Player:
    def __init__(self, name: str, tag: str, age: int, pronouns: int, region: int, advantage: int, disadvantage: int, neutral: int, mental: int):
        self.name = name
        self.tag = tag
        self.age = age
        self.pronouns = pronouns
        self.region = region
        self.advatange = advantage
        self.disadvantage = disadvantage
        self.neutral = neutral
        self.mental = mental
        #self.main = main

# Only toggle this for quit options basically
game_running = True

# variables that need to be accessed by multiple files.
player_name = ""
player_tag = ""
player_pronouns = -1
player_region = -1

player = Player(player_name, player_tag, 13, player_pronouns, player_region, 0, 0, 0, 0)

# these are more game state instead of player state


# format: {region name, difficulty name, diffculty num}
Regions = {
    1: ["Japan", "Very Competitive", 5],
    2: ["NA East", "Competitive", 4],
    3: ["China", "Competitive", 4],
    4: ["UK/Ireland", "Hard", 4],
    5: ["South Korea", "Hard", 4],
    6: ["EU West", "Moderate", 3],
    7: ["NA West", "Moderate", 3],
    8: ["NA Midwest", "Moderate", 3],
    9: ["South America", "Moderate", 3],
    10: ["Europe North/East", "Light", 2],
    11: ["Mexico", "Light", 2],
    12: ["Middle East", "Light", 2],
    13: ["Asia Southeast", "Obscure", 1],
    14: ["Asia South", "Obscure", 1],
    15: ["Central America", "Obscure", 1],
    16: ["Africa", "Obscure", 1],
    17: ["Oceaina", "Obscure", 1]
}

Pronouns = {
    1: ["he", "him"],
    2: ["she", "her"],
    3: ["they", "them"] 
}

Game_Characters = {
    # TODO:
    # Add characters, selecting a main, and tier lists of characters that change with meta
    # Counters are less important because it is less prevalent in FGC games
}



# neutral, advantage, disadvantage is the basics. The hero select is mainly aesthetic but necessary I think. 
# 
        